dbt_sugar/core/ui/cli_ui.py

```diff
@@ -169,13 +173,57 @@ def _validate_question_payload(self) -> None:
         self._is_valid_question_payload = True
 
     @staticmethod
-    def _iterate_through_columns(cols: List[str]) -> Dict[str, str]:
+    def _iterate_through_columns(
+        cols: List[str], ask_for_tests: bool = True
+    ) -> Mapping[str, Mapping[str, Union[str, List[str]]]]:
+        """Iterates through a provided list of columns collects documentation info.
+
+        More specifically it asks for the following
+        - a column documentation
+        - whether user wants to add tests
+            - and if so which tests to add from a list of pre-defined/aggreed tests.
+
+        Args:
+            cols (List[str]): List of columns to collect description and tests info from.
+            ask_for_tests (bool, optional): When set to true UI will ask for whether the user wants
+            to add tests for each of the columns. Defaults to False.
+
+        Returns:
+            Mapping[str, Mapping[str, Union[str, List[str]]]]: API paylad to be passed to back-end
+            that looks like this:
+            ```python
+            {
+                'col_a': {
+                    'description': 'Description for col a',
+                    'tests': ['unique']
+                    },
+                'col_b': {'description': 'Description for col b'}
+            }
+            ```
+        """
         results = dict()
         for column in cols:
             description = questionary.text(message=f"{column}: {DESCRIPTION_PROMPT_MESSAGE}").ask()
-            if description:
-                results.update({column: description})
 
+            if description:
+                results.update({column: {"description": description}})
+            else:
+                # we still want to make the column end up in the dict but we don't put a desc if
+                # empty... not sure about this
+                results.update({column: {}})
+
```

Comment:
I think is better, if there is no description we don't add them into results (we don't do any action with that column).

Reply:
Ah at first I agreed with this but I cannot do that here. 

I think that there may be a case where someone doesn't want to add a description but they might still want to add a test. In that case we will want to have the column added here.

However, what I can do **before** I return it to you is that if the dict for that column is still empty (because they didn't add a description and they didn't add a test) then I could pick it out of the results. Would that make sense?

Reply:
yes, is fine! I have adapted the update to not update when is None.

Reply:
I'm not sure I understand

Reply:
I ended up implementing what I proposed at the end of https://github.com/bitpicky/dbt-sugar/pull/68#discussion_r567424371 just in case I think it's cleaner.